homeassistant/components/volvooncall/binary_sensor.py

```diff
@@ -16,8 +16,11 @@ class VolvoSensor(VolvoEntity, BinarySensorEntity):
 
     @property
     def is_on(self):
-        """Return True if the binary sensor is on."""
-        return self.instrument.is_on
+        """Return True if the binary sensor is on, but invert for the 'Door lock'."""
+        if self.instrument.name == "Door lock":
```

Comment:
Are there other Volvo locks (other than door locks) that don't behave like this or do all locks set `instrument.is_on` to True when they are locked?

Reply:
This is the only lock (on my Volvo anyway). It exists as a switch that you can toggle remotely (the Lock component), and as a binary sensor. Weirdly, only the binary sensor was inverted. The lock/switch component already worked fine. Also all the other binary_sensors (open/closed state of all the door and windows) were fine. I guess Volvo just defined this specific sensor the other way around.
Does that answer your question?

Reply:
If all volvo locks are define like this I think it's ok to fix this here. If it's just this type of lock, it should preferably be fixed in the library so that they all behave the same.

Reply:
I'm sorry; I'm not really following you. (This is probably due to me being very new to coding for home assistant.)
Do we agree that, since this is the only lock binary_sensor, it is ok like this? (I really wouldn't know how to do it differently.)

Reply:
Yes, if you can confirm that that's the case. Ie, there are no other Volvo binary sensor locks.

Reply:
Ok, great. I can confirm that!

Reply:
Let's check the `attr` attribute instead of the `name` attribute.
```suggestion
        if self.instrument.attr == "is_locked":
```

Reply:
Tested, that works too. And as expected doesn't break anything else.